xarray/conventions/netcdftimeindex.py

```diff
@@ -120,23 +118,31 @@ def get_date_type(self):
     return type(self._data[0])
 
 
-def assert_all_same_netcdftime_datetimes(data):
-    from netcdftime._netcdftime import datetime
+def assert_all_valid_date_type(data):
+    from netcdftime import (
```

Comment:
Sorry, I buried this in a comment (https://github.com/pydata/xarray/pull/1252#discussion_r100678111) above.  Confusingly, ```netcdftime.datetime``` does not refer to the super class:
```
In [1]: from netcdftime import datetime, DatetimeAllLeap

In [2]: datetime(1, 1, 1)
Out[2]: netcdftime._netcdftime.DatetimeProlepticGregorian(1, 1, 1, 0, 0, 0, 0, -1, 1)

In [3]: test = DatetimeAllLeap(1, 1, 1)

In [4]: isinstance(test, datetime)
Out[4]: False

In [5]: from netcdftime._netcdftime import datetime as super_datetime

In [6]: isinstance(test, super_datetime)
Out[6]: True
```